Setup vagrant for Go Development

aludvik opened the pull request:
Test that everything (go, protogen, protobuf, zmq) is working by provisioning vagrant and then:
1. `protogen go`
2. `cd sdk/go/src/sawtooth_sdk`
3. `go run fake_validator.go`
4. `go run fake_processor.go`

peterschwarz commented on protos/batch.proto:
```diff
@@ -17,6 +17,7 @@ syntax = "proto3";
 
 option java_multiple_files = true;
 option java_package = "sawtooth.sdk.protobuf";
+option go_package = "sawtooth_sdk/protobuf";
```
This commit should probably be sorted before the protogen changes.

aludvik replied:
Hmm, can you do `git log` locally? I am seeing a discrepancy between the ordering here and local.

aludvik replied:
Or you can see the ordering here: https://github.com/aludvik/sawtooth-core/commits/install_go

vaporos replied:
The go_package should perhaps include the proto name; for example "sawtooth_sdk/protobuf/batch", since having these all in the same namespace is awkward.

aludvik replied:
I tried this first, but was getting a lot of complaints from Go's protoc plugin. Will try again

aludvik commented:
Added install_go_tools to default plugin list.

vaporos commented on tools/plugins/install_go_tools.sh:
```diff
@@ -0,0 +1,23 @@
+echo "export GOPATH=$GOPATH:/project/sawtooth-core/sdk/go" >> /home/$VAGRANT_USER/.bashrc
+echo "export PATH=$PATH:$GOBIN" >> /home/$VAGRANT_USER/.bashrc
+
+apt-get install -y -q \
```
This should be installed in the package_groups/ directory instead of a manual apt-get.

aludvik replied:
Should all the other apt-get statements for the other plugins be moved there as well? Creating new plugins has been the pattern I've been following for all additions, but maybe that was incorrect.

vaporos commented on tools/plugins/install_go_tools.sh:
```diff
@@ -0,0 +1,23 @@
+
+export GOPATH
+echo "export GOPATH=$GOPATH:/project/sawtooth-core/sdk/go" >> /home/$VAGRANT_USER/.bashrc
+echo "export PATH=$PATH:$GOBIN" >> /home/$VAGRANT_USER/.bashrc
```
This clobbers PATH since $PATH is expanded here.  Suggest adding GOPATH, PATH statements to local-env.sh instead.

vaporos commented on sdk/go/src/sawtooth_sdk/fake_validator.go:
```diff
@@ -0,0 +1,45 @@
+package main
```
No need to commit a fake validator.  The fake processor works fine against the real validator if the validator is patched and the request is enclosed in a Message.

vaporos commented on sdk/go/src/sawtooth_sdk/fake_processor.go:
```diff
@@ -0,0 +1,44 @@
+	socket.Connect(endpoint)
+	fmt.Println("done")
+
+	// Send registration request
```
If we keep this file in this PR, let's update it to send a message the validator can understand.  Something like:

```	// Send registration request
	request := &protobuf.TpRegisterRequest{
		Family:     "test_family",
		Version:    "1.0",
		Encoding:   "protobuf",
		Namespaces: []string{"test_namespace"},
	}
	requestData, err := proto.Marshal(request)
	if err != nil {
		fmt.Printf("Failed to marshal: %v\n", err)
	}

	correlationId := "123"
	msg := &protobuf.Message{
		MessageType: protobuf.Message_TP_REGISTER_REQUEST,
		CorrelationId: correlationId,
		Content: requestData,
	}

	msgData, err := proto.Marshal(msg)
	if err != nil {
		fmt.Printf("Failed to marshal: %v\n", err)
	}
```

aludvik commented:
1. Fixed vagrant setup.
2. Created separate packages for each .proto file
3. Changed fake_processor.go -> noop_processor.go which tries to register and unregister with a validator, but it doesn't totally work yet.
